### PR TITLE
Bind Contextvars before metadata is parsed

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?
Moves the bind of context vars before the parsing of Metadata, allowing us to have additional context when checking logs when the Metadata parser step has failed.

### How to review
Launch a schema with invalid metadata and check that relevant vars have been bound to the logs.

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by berroar** [comment]
[Bind context to logs before metadata parser runs (1 SP)](https://trello.com/c/plhQCbsA/5784-bind-context-to-logs-before-metadata-parser-runs-1-sp)

File: app/routes/session.py [hunk 1]
```diff
@@ -73,6 +73,20 @@ def login() -> Response:
...
     validate_jti(decrypted_token)
 
+    logger_args = {
+        key: value
+        for key, value in {
+            "tx_id": decrypted_token.get("tx_id"),
+            "case_id": decrypted_token.get("case_id"),
+            "schema_name": decrypted_token.get("schema_name"),
+            "schema_url": decrypted_token.get("schema_url"),
+            "ru_ref": decrypted_token.get("ru_ref"),
```
Comment (katie-gardner) [inline app/routes/session.py:83]: in the case of auth payload v2, I I think `ru_ref` will be under `survey_metadata`
so I'm seeing it in debug as `decrypted_token["survey_metadata"]["data"]["ru_ref"]` and maybe same for qid? I think you might need an if else/ternary for that case

**Review by katie-gardner — CHANGES_REQUESTED** [review]
If I launch with v2 I'm not getting `ru_ref` or `qid` bound, others are ok though

File: app/routes/session.py (hunk 1 above)
Comment (berroar) [reply app/routes/session.py]: Updated now

Comment (katie-gardner) [reply app/routes/session.py]: the update still isn't quite working as it has changed to `decrypted_token.get("survey_metadata", {}).get("ru_ref")`
but it needs to be `decrypted_token.get("survey_metadata", {}).get("data", {}).get("ru_ref")`

**Review by katie-gardner — APPROVED** [review]
Now working as I'd expect for v1 and v2 👍

**Review by petechd — APPROVED** [review]

File: app/routes/session.py [hunk 2]
```diff
@@ -73,6 +73,32 @@ def login() -> Response:
...
+        decrypted_token.get("survey_metadata", {}).get("data", {}).get("ru_ref")
+        if decrypted_token.get("survey_metadata")
+        else decrypted_token.get("ru_ref")
+    )
+
+    qid = (
+        decrypted_token.get("survey_metadata", {}).get("data", {}).get("qid")
+        if decrypted_token.get("survey_metadata")
+        else decrypted_token.get("qid")
+    )
```
Comment (MebinAbraham) [inline app/routes/session.py:86]: I suggest simplifying this to improve readability/reduce duplication. You can assign survey metadata for reuse.

Either a simple if else.
```
_data = (
    survey_metadata.get("data", {})
    if (survey_metadata := decrypted_token.get("survey_metadata"))
    else decrypted_token
)
ru_ref, qid = _data.get("ru_ref"), _data.get("qid")
```

Comment (berroar) [reply app/routes/session.py]: Updated 👍

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**